dynamiq/components/retrievers/pgvector.py

```diff
@@ -0,0 +1,74 @@
+from typing import Any
+
+from dynamiq.storages.vector import PGVectorStore
+from dynamiq.utils.logger import logger
+
+
+class PGVectorDocumentRetriever:
+    """
+    Document Retriever using PGVector.
+    """
+
+    def __init__(
+        self,
+        *,
+        vector_store: PGVectorStore,
+        filters: dict[str, Any] | None = None,
+        top_k: int = 10,
+    ):
+        """
+        Initializes a component for retrieving documents from a PGVector vector store with optional filtering.
+
+        Args:
+            vector_store (PGVectorStore): An instance of PGVectorStore to interface with PGVector vectors.
+            filters (Optional[dict[str, Any]]): Filters to apply for retrieving specific documents. Defaults to None.
+            top_k (int): The maximum number of documents to return. Defaults to 10.
+
+        Raises:
+            ValueError: If the `vector_store` is not an instance of `PGVectorStore`.
+
+        This initializer checks if the `vector_store` provided is an instance of the expected `PGVectorStore`
+        class, sets up filtering conditions if any, and defines how many top results to retrieve in document queries.
+        """
+        if not isinstance(vector_store, PGVectorStore):
+            msg = "document_store must be an instance of PGVectorStore"
+            raise ValueError(msg)
+
+        self.vector_store = vector_store
+        self.filters = filters or {}
+        self.top_k = top_k
+
+    def run(
+        self,
+        query_embedding: list[float],
+        exclude_document_embeddings: bool = True,
+        top_k: int | None = None,
+        filters: dict[str, Any] | None = None,
```

Comment:
Please check improvements from this one PR: https://github.com/dynamiq-ai/dynamiq/pull/70.
Need to add support of
```
content_key: str | None = None,
embedding_key: str | None = None,
```